File: Largest Combination With Bitwise AND Greater Than Zero.go
func largestCombination(candidates []int) int {
    dp := make([]int, 0)
    countBits := func(num int) {
        i := 0
        for num > 0 {
            if i == len(dp) {
                dp = append(dp, 0)
            }
            dp[i] += num % 2
            i++
            num >>= 1
        }
    }

    for _, num := range candidates {
        countBits(num)
    }
    return slices.Max(dp)
}

File: Largest Combination With Bitwise AND Greater Than Zero.py
class Solution:
    def largestCombination(self, candidates: List[int]) -> int:
        dp = []

        def countBits(num):
            nonlocal dp
            i = 0
            while num:
                if i == len(dp):
                    dp.append(0)
                dp[i] += num % 2
                i += 1
                num >>= 1

        for num in candidates:
            countBits(num)
        return max(dp)
